File: day1/day1.lua
-- Solve puzzel 1 
function puzzel1()
    local file = io.open("day1_data.txt")
    local freq = 0
    for line in file:lines("*number") do 
        freq = freq + line
    end
    print(freq)
    file:close()
end


function puzzel2()
    local file = io.open("day1_data.txt")
    local changes = {}
    for line in file:lines("*number") do 
        table.insert(changes, line)
    end

    local freq = 0 
    local idx = 1
    local freqs = {}
    local notFound = true
    while notFound do
        freq = freq + changes[idx]

        for i, val in ipairs(freqs) do
            if freq == val then
                notFound = false
                print(freq)
                break
                
            --elseif freq > val then
            --    table.insert( freqs, i+1, freq)
            --    break
            end
        end
        table.insert( freqs,freq)

        if idx == #changes then
            idx = 1
        else
            idx = idx + 1
        end
    end
end

puzzel1()
puzzel2()

File: day1/day1.py

def puzzel1():
    freq = 0
    with open("day1_data.txt") as f:
        for line in f:
            freq += int(line)
    print(freq)

def puzzel2():
    with open("day1_data.txt") as f:
        changes = [int(line) for line in f]
    
    same_freq_found = False
    freqs = [changes[0]]
    freq = changes[0]
    idx = 1
    while not same_freq_found:
        if idx == len(changes):
            idx = 0
        freq += changes[idx]
        if freq in freqs:
            print(freq)
            same_freq_found = True
            break
        else:
            freqs.append(freq)
            idx += 1

def puzzel1_short_solution():
    print(sum([int(line) for line in open("day1_data.txt")]))


if __name__ == "__main__":
    puzzel1()
    puzzel1_short_solution()
    puzzel2()